.github/workflows/ci.yml

```diff
@@ -31,22 +25,18 @@ jobs:
       uses: actions/cache@v2
       with:
         path: vendor/bundle
-        key: "${{ runner.OS }}-gem-cache-${{ hashFiles('**/Gemfile.lock') }}"
+        key: "${{ runner.OS }}-gem-cache-${{ hashFiles('**/Gemfile.lock')
+          }}"
         restore-keys: "${{ runner.OS }}-gem-cache-\n"
-    - name: Install bundler
-      run: (bundler -v | grep "2.1.4") || gem install bundler:2.1.4
-    - name: Remove metrics from bundled gems
-      run: bundle config set without 'metrics'
-    - name: Install gems
-      run: bundle install --jobs $(nproc) --retry 3 --deployment
-    - name: Create cache directory
-      run: mkdir -p tmp/cache
-    - name: Load database structure
-      run: bundle exec rake db:create db:schema:load
-    - name: Run RSpec test suite
-      run: bundle exec rspec spec
-      env:
-        METRICS: '1'
+    - uses: "cachix/install-nix-action@8d6d5e949675fbadb765c6b1a975047fa5f09b27"
+      with:
+        extra_nix_config: |
+          post-build-hook = /etc/nix/upload-to-cache.sh
+          substituters = https://cache.nixos.org/
+          trusted-public-keys = cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
+        nix_path: nixpkgs=channel:nixos-21.05
+    - name: Run CI through nix-shell
+      run: nix-shell --run "chmod 755 ./run_ci.sh && ./run_ci.sh"
```

Comment:
You should be able to set `+x` on the script and commit it to the repo?

Reply:
I could, but what's the harm of doing it here so people don't have to remember to do that?

Reply:
I just think it's cleaner but there's no harm.